packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -325,6 +327,10 @@ export default function lwc(pluginOptions: RollupLwcOptions = {}): Plugin {
                 scopedStyles: scoped,
                 disableSyntheticShadowSupport,
                 apiVersion: apiVersionToUse,
+                // Only pass this in if it's actually specified – otherwise unspecified becomes undefined becomes false
+                ...('enableStaticContentOptimization' in pluginOptions && {
+                    enableStaticContentOptimization: pluginOptions.enableStaticContentOptimization,
+                }),
```

Comment:
This is the only functional change in this PR.

I personally find it weird that `undefined` means `false` but "unspecified" means `true`, but this is how `@lwc/template-compiler` and `@lwc/compiler` already work, so I wanted `@lwc/rollup-plugin` to work the same.

I considered adding a new option, `disableStaticContentOptimization`, but I figured this would just confuse things further. Especially because this is mostly just used for our own Karma tests.